### DISCUSSION TITLE
Troubleshooting/debugging when updating Quarto website on Netlify

### DISCUSSION THREAD

**Discussion by giocomai** [quarto-dev/quarto-cli#3242] [Q&A]
I have been able to successfully publish a website via Netlify, all went smoothly.

However, if I update some files, render and run again `quarto publish netlify` I get:

```
? Publish update to:
❯ https://whatever-word-a12345.netlify.app (Netlify)
  Add a new destination...
```

And then nothing. It just remains there, even if this is a tiny website. I tried again with a fresh project in a fresh folder, and got the same: all smooth at first upload and then just stuck with no information when updating. 

My main issue here is perhaps the fact that I do not know how to troubleshoot this further, as I would do if this was in R. 

All parameters in `_publish.yml" look exactly as in the documentation, and anyway, this happens even with a freshly created project from Rstudio with no other action taken but. "quarto publish netlify", change a small thing in the website, and run the same command again (a website on netlify is generated at first run, but then it gets stuck if I try to update).

So in a way, the question is both specific about possible ways to deal with it, as well as more generically about troubleshooting: what do I do if Quarto is just stuck and doesn't show any error? is there some "--verbose" flag I am missing? I couldn't find a troubleshooting section in the otherwise absolutely fantastic documentation, but I suppose I'm missing something basic here. 

Thank you for any assistance with this.

For reference, here's my `quarto check` (updated to latest version while troubleshooting, had same problem with older version)
<details>

```

[✓] Checking Quarto installation......OK
      Version: 1.2.267
      Path: /home/g/opt/quarto-1.2.267/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.7
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.1.3
      Path: /usr/lib64/R
      LibPaths:
        - /home/g/R/x86_64-redhat-linux-gnu-library/4.1
        - /usr/lib64/R/library
        - /usr/share/R/library
      rmarkdown: 2.17

[✓] Checking Knitr engine render......OK

```

</details>

**Comment by ndillmanhasso** [comment]
Did you figure this out? I'm having the same problem and I noticed it happened when I changed the netlify domain...

**Reply by giocomai** [reply]
no, actually, I've been updating manually on Netlify via the web interface, which kind of defeats the purpose of using Netlify, so I've been considering changing workflow / use a different service. But it still feels like something that *should* be easy to fix, if one could just see what the issue is or where exactly it gets stuck.

You mention this happened when you changed domain on Netlify... I also may have introduced a custom domain only after the first upload. But to me, as described above, this happens also with new projects / new websites that have no custom domain. Perhaps the name change leaves something strange in the local cache that messes up the whole thing? 

Or perhaps it's something specific to the operating system? I'm on Fedora Linux, and even if I don't see how exactly how, I wonder if there may be something there.

**Comment by giocomai** [comment]
Troubleshooting this further.

If I go in my `/home/user/.local/share/quarto/publish/accounts/` (change user to your user), I saw that I had two folders:
- `netlify` (all lower case)
- `Netlify` (capital first letter)

Only `Netlify` had an "accounts.json" file in it, the other was empty.

Even if I have a set up a custom domain, when I did `quarto publish` I still saw the process starting as publishing the update to:
`https://whatever-word-a12345.netlify.app (Netlify)`

If I tried to copy the "accounts.json" file from `Netlify` to the `netlify` subfolder, however, it appeared it started publishing to my custom domain. 

But it still get stuck. 

If I delete both subfolders, it then recreates only the `netlify` subfolder (all lowercase), but stores nothing in it, and when it uploads, it appears it is starting to upload to  `whatever-word-a12345.netlify.app`, not my custom domain. Also, somewhat puzzingly, it does not ask me to authenticate, which at this point I think it should. 

It seems to me there may be something about file ownership / access rights that is not quite right.

**Comment by giocomai** [comment]
Actually, this is where I things go off the rails. When I run `quarto publish` from the terminal, the first time I get the prompt from the operating system asking for access to one of my local ssh keys. But why is it needed? I thought `quarto` got its token from `netlify`. 

So if I run `quarto publish`, first thing I get this prompt:

![Screenshot from 2022-11-29 09-27-19-obfuscated](https://user-images.githubusercontent.com/11276752/204477987-ace0f648-28c2-40ac-a34f-3bc2eb64d026.png)

Then no matter if I give my password or hit on "cancel", I see on the terminal:
`? Publish update to:`
`❯ ...` [address]

Either way, nothing is written in the newly created `/home/user/.local/share/quarto/publish/accounts/netlify/` folder

**Comment by rongbinf** [comment]
**Try hitting `enter` after `quarto publish netlify`**. It might be an option for users with multiple Netlify sites.

**Reply by giocomai** [reply]
hilariously, this is the right answer. I've spent at least a couple of hours troubleshooting this, trying to figure out internals of various components, and this was it: just hit enter.

Admittedly, I didn't recognise it as a prompt, as I haven't seen anything quite like it in the console. My bad, I suppose. 

Thanks a bunch.
